src/Books/Application/BooksFinder.ts

```diff
@@ -1,11 +1,23 @@
 import {Book} from "../Domain/Book";
 import {BookRepository} from "../Domain/BookRepository";
+import {GetBookControllerResponse} from "../Domain/GetBookControllerResponse";
 
 export class BooksFinder {
     constructor(private repository: BookRepository) {}
 
-    async run(): Promise<Book[]> {
-        return this.repository.findAll();
+    async run(): Promise<GetBookControllerResponse[]> {
+        const books = this.repository.findAll();
+
+        let res: GetBookControllerResponse[] = [];
+
+        books.forEach(function (book: Book) {
+            res.push(new GetBookControllerResponse(
+                book.id.uuid,
+                book.name.value,
+                book.length.value))
+        });
```

Comment:
This part is correct. However, it is better to use arrow functions instead of function because if we follow this policy (use always arrow functions) some parts of the code need less words to do the same, making it easier to read and understand.

 `books.forEach( book: Book => {
              (...)
          });`